src/compiler/checker.ts

```diff
@@ -4820,12 +4820,21 @@ namespace ts {
         }
 
         function getInstantiatedConstructorsForTypeArguments(type: Type, typeArgumentNodes: TypeNode[], location: Node): Signature[] {
-            let signatures = getConstructorsForTypeArguments(type, typeArgumentNodes, location);
-            if (typeArgumentNodes) {
+            const signatures = getConstructorsForTypeArguments(type, typeArgumentNodes, location);
+            if (some(signatures)) {
+                const result: Signature[] = [];
                 const typeArguments = map(typeArgumentNodes, getTypeFromTypeNode);
-                signatures = map(signatures, sig => getSignatureInstantiation(sig, typeArguments));
+                for (const sig of signatures) {
+                    if (some(sig.typeParameters)) {
+                        result.push(getSignatureInstantiation(sig, typeArguments));
+                    }
+                    else {
+                        result.push(sig);
+                    }
+                }

```

Comment:
Even better, use `sameMap` and save an allocation when none of them need to be instantiated (which is often the case).